src/web/js/beforePyret.js

```diff
@@ -1,48 +1,55 @@
-LOG = true;
-function ct_log(/* varargs */) {
+/* global $ jQuery CPO CodeMirror storageAPI Q createProgramCollectionAPI makeShareAPI */
+
+var shareAPI = makeShareAPI(process.env.CURRENT_PYRET_RELEASE);
+
+var url = require('url.js');
+
+const LOG = true;
+window.ct_log = function(/* varargs */) {
   if (window.console && LOG) {
     console.log.apply(console, arguments);
   }
-}
-function ct_error(/* varargs */) {
+};
+
+window.ct_error = function(/* varargs */) {
   if (window.console && LOG) {
     console.error.apply(console, arguments);
   }
-}
+};
 var initialParams = url.parse(document.location.href);
 var params = url.parse("/?" + initialParams["hash"]);
-var highlightMode = "mcmh";
+var highlightMode = "mcmh"; // what is this for?
```

Comment:
@jswrenn ping (I know the answer—it corresponds to the highlighting choices in the dropdown—but Jack can probably answer follow-on questions better)